thrust/system/cuda/detail/trivial_copy.inl

```diff
@@ -66,6 +66,16 @@ template<typename System1,
   return cudaMemcpyHostToDevice;
 } // end cuda_memcpy_kind()
 
+#if defined(_WIN32) && !defined(_WIN64)
```

Comment:
Not quite: CUDA supports 32-bit mode only on Windows, ergo windows-specific fix.

We can't remove #if because of the  possible use  thrust::copy(thrust::cuda::par, raw_host_ptr, raw_host_ptr+n, raw_device_ptr). The cudaMemcpyDefault in 64-bit will figure out correct src/destnation memory space and will do host->device, copy.